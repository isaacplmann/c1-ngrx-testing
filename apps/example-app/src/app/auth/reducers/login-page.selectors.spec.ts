import {
  authFeatureKey,
  AuthState,
  selectAuthState,
  State,
  selectUser,
  selectLoggedIn
} from './';

function createAuthState(): State {
  return <any>{
    [authFeatureKey]: {
      loginPage: {
        pending: false,
        error: undefined
      },
      status: {
        user: {
          name: 'someone'
        }
      }
    }
  };
}

describe('LoginPageSelectors', () => {
  it('should test selectors', () => {
    const state = createAuthState();
    expect(selectAuthState(state)).toBe(state[authFeatureKey]);
    expect(selectUser(state)).toBe(state[authFeatureKey].status.user);
    expect(selectLoggedIn(state)).toBe(true);
  });

  const testCases = [
    {
      name: 'selectAuthState',
      selector: selectAuthState,
      state: createAuthState()
    },
    {
      name: 'selectUser',
      selector: selectUser,
      state: createAuthState()
    },
    {
      name: 'selectLoggedIn',
      selector: selectLoggedIn,
      state: createAuthState()
    }
  ];
  testCases.forEach(({ name, state, selector }) => {
    it(`should test ${name} with input ${JSON.stringify(state)}`, () => {
      expect(selector(state)).toMatchSnapshot();
    });
  });

  // using projector
  it('should test just the selectLoggedIn logic', () => {
    expect(selectLoggedIn.projector({ name: 'something' })).toBe(true);
    expect(selectLoggedIn.projector(undefined)).toBe(false);
  });
});
