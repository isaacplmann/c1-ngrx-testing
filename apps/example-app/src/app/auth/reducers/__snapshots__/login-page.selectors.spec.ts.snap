// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPageSelectors should test selectAuthState with input {"auth":{"loginPage":{"pending":false},"status":{"user":{"name":"someone"}}}} 1`] = `
Object {
  "loginPage": Object {
    "error": undefined,
    "pending": false,
  },
  "status": Object {
    "user": Object {
      "name": "someone",
    },
  },
}
`;

exports[`LoginPageSelectors should test selectLoggedIn with input {"auth":{"loginPage":{"pending":false},"status":{"user":{"name":"someone"}}}} 1`] = `true`;

exports[`LoginPageSelectors should test selectUser with input {"auth":{"loginPage":{"pending":false},"status":{"user":{"name":"someone"}}}} 1`] = `
Object {
  "name": "someone",
}
`;
